### PR TITLE
Harden Plugins::ManagerBase::update() against exceptions.

### PR DESCRIPTION
**Pull request by bangerth**
This is taken from how `MeshRefinement::Manager::update()` does things. I'm just moving that into the new `Plugins::ManagerBase` class.

Part of #1775.

### CODE REVIEW

File: include/aspect/plugins.h [hunk 1]
```diff
@@ -262,9 +262,56 @@ namespace aspect
     template <typename InterfaceType>
     void ManagerBase<InterfaceType>::update()
     {
-      for (const auto &plugin : plugin_objects)
+      // call the update() functions of all refinement plugins.
```
Comment (gassmoeller) [inline include/aspect/plugins.h:265]:
```suggestion
      // call the update() functions of all plugins.
```

File: include/aspect/plugins.h [hunk 2]
```diff
@@ -262,9 +262,56 @@ namespace aspect
...
+              MPI_Abort (MPI_COMM_WORLD, 1);
+            }
+          catch (...)
+            {
+              std::cerr << std::endl << std::endl
+                        << "----------------------------------------------------"
+                        << std::endl;
+              std::cerr << "Exception on MPI process <"
+                        << dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)
+                        << "> while running mesh refinement plugin <"
```
Comment (gassmoeller) [inline include/aspect/plugins.h:304]:
```suggestion
                        << "> while running plugin <"
```

File: include/aspect/plugins.h [hunk 3]
```diff
@@ -262,9 +262,56 @@ namespace aspect
...
+          // an exception is generated, catch it, print an error message,
+          // and abort the program
+          catch (std::exception &exc)
+            {
+              std::cerr << std::endl << std::endl
+                        << "----------------------------------------------------"
+                        << std::endl;
+              std::cerr << "Exception on MPI process <"
+                        << dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)
+                        << "> while running mesh refinement plugin <"
```
Comment (gassmoeller) [inline include/aspect/plugins.h:286]:
```suggestion
                        << "> while running plugin <"
```

**Review by gassmoeller — APPROVED** [review]
Nice that this extends to all managers now. Good to go with the minor changes.

**Comment by bangerth** [comment]
Oh, doh! Thanks for carefully reading what I wrote!

**Comment by gassmoeller** [comment]
looks like that affected some tests

**Comment by bangerth** [comment]
Yes, I updated those just now.

**Pull request merged**